torch/_meta_registrations.py

```diff
@@ -4766,24 +4784,46 @@ def meta__scaled_dot_product_flash_backward(
     num_heads = query.size(1)
     head_dim = query.size(3)
 
-    grad_q = torch.empty_permuted(
-        (batch_size, num_heads, max_q, head_dim),
-        (0, 2, 1, 3),
-        dtype=query.dtype,
-        device=query.device,
-    )
-    grad_k = torch.empty_permuted(
-        (batch_size, num_heads, max_k, head_dim),
-        (0, 2, 1, 3),
-        dtype=key.dtype,
-        device=key.device,
-    )
-    grad_v = torch.empty_permuted(
-        (batch_size, num_heads, max_k, head_dim),
-        (0, 2, 1, 3),
-        dtype=value.dtype,
-        device=value.device,
-    )
+    if device_hint(query) == "cpu":
+        seq_q = query.size(2)
+        seq_kv = value.size(2)
+        grad_q = torch.empty_permuted(
+            (batch_size, num_heads, seq_q, head_dim),
+            (0, 2, 1, 3),
+            dtype=query.dtype,
+            device=query.device,
+        )
+        grad_k = torch.empty_permuted(
+            (batch_size, num_heads, seq_kv, head_dim),
+            (0, 2, 1, 3),
+            dtype=key.dtype,
+            device=key.device,
+        )
+        grad_v = torch.empty_permuted(
+            (batch_size, num_heads, seq_kv, head_dim),
+            (0, 2, 1, 3),
+            dtype=value.dtype,
+            device=value.device,
+        )
```

Comment:
I think we can further simplify the code if the differences are only the seq length.

Reply:
Sure! Modified.